src/liballoc/btree/node.rs

```diff
@@ -168,6 +190,25 @@ unsafe impl<K: Sync, V: Sync> Sync for Root<K, V> { }
 unsafe impl<K: Send, V: Send> Send for Root<K, V> { }
 
 impl<K, V> Root<K, V> {
+    pub fn is_shared_root(&self) -> bool {
+        self.as_ref().is_shared_root()
+    }
+
+    pub fn shared_empty_root() -> Self {
+        // Ensuring that the shared node hasn't been corrupted by any mutations
+        debug_assert!(EMPTY_ROOT_NODE.parent == ptr::null());
+        debug_assert!(EMPTY_ROOT_NODE.parent_idx == 0);
+        debug_assert!(EMPTY_ROOT_NODE.len == 0);
```

Comment:
Is this even possible? It should literally be in rodata...

Reply:
Oh, right, that'd be why I was getting segfaults when it got modified. I'll remove this.